File: 4 两个排序数组的中位数.c
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> numsAll;
        
        for (int i = 0, j = 0; i < nums1.size() || j < nums2.size();) {
            if (i < nums1.size()) {
                if (j < nums2.size()) {
                    if (nums1[i] < nums2[j]) {
                        numsAll.push_back(nums1[i++]);
                    } else {
                        numsAll.push_back(nums2[j++]);
                    }
                } else {
                    numsAll.push_back(nums1[i++]);
                }
            } else {
                numsAll.push_back(nums2[j++]);
            }
        }
        
        int middle = (nums1.size() + nums2.size()) >> 1;
        if ((nums1.size() + nums2.size()) % 2 == 1) {
            return (double)numsAll[middle];
        } else {
            return ((double)numsAll[middle] + (double)numsAll[middle-1]) / 2.0;
        }
    }
};

File: 4 两个排序数组的中位数.java
public class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int length = nums1.length + nums2.length;
        if (length%2 == 1){
            return recrusive(nums1, nums2, 0, nums1.length-1, 0, nums2.length-1, length/2 + 1);
        } else {
            return (recrusive(nums1, nums2, 0, nums1.length-1, 0, nums2.length-1, length/2)+recrusive(nums1, nums2, 0, nums1.length-1, 0, nums2.length-1, length/2 + 1))/2.0;
        }
    }
    
    public double recrusive(int[] A, int[] B, int i, int i2, int j, int j2, int k) {
        int m = i2-i+1;
        int n = j2-j+1;
        if (m > n) return recrusive(B, A, j, j2, i, i2, k);
        if (m == 0) {
            return B[j+k-1];
        }
        if (k == 1) {
            return Math.min(A[i], B[j]);
        }
        int posA = Math.min(k/2, m);
        int posB = k-posA;
        if(A[i+posA-1] == B[j+posB-1]) {
            return A[i+posA-1];
        } else if (A[i+posA-1] < B[j+posB-1]) {
            return recrusive(A, B, i+posA, i2, j, j+posB-1, k-posA);
        } else {
            return recrusive(A, B, i, i+posA-1, j+posB, j2, k-posB);
        }
    }
}
